fastify.js

```diff
@@ -97,7 +97,9 @@ function build (options) {
 
   var server
   const httpHandler = router.lookup.bind(router)
-  if (options.https) {
+  if (options.serverFactory) {
+    server = options.serverFactory(httpHandler)
+  } else if (options.https) {
```

Comment:
A better implementation of this code is to have some server factories around and select one based on the options.

Reply:
You mean as a strategy pattern implementation?

Reply:
ssshhh patterns are not cool anymore ;).